File: be/src/main/java/sg/flow/services/TransferServices/TransferService.java
package sg.flow.services.TransferServices;

import java.util.List;

import sg.flow.entities.Bank;
import sg.flow.models.transfer.TransferRecepient;
import sg.flow.models.transfer.TransferRequestBody;
import sg.flow.models.transfer.TransferResult;

public interface TransferService {

    TransferResult sendTransaction(TransferRequestBody sendRequestBody);

    List<TransferRecepient> getRelevantRecepient();

    List<Bank> getRelevantRecepientByAccountNumber(String accountNumber);

    TransferRecepient getRelevantRecepientByContact(String contact);
    
}

File: be/src/main/kotlin/sg/flow/services/TransferServices/TransferService.kt
package sg.flow.services.TransferServices

import sg.flow.entities.Bank
import sg.flow.models.transfer.TransferRecepient
import sg.flow.models.transfer.TransferRequestBody
import sg.flow.models.transfer.TransferResult

interface TransferService {
    suspend fun sendTransaction(sendRequestBody: TransferRequestBody): TransferResult
    suspend fun getRelevantRecepient(): List<TransferRecepient>
    suspend fun getRelevantRecepientByAccountNumber(accountNumber: String): List<Bank>
    suspend fun getRelevantRecepientByContact(contact: String): TransferRecepient
}
